Framework/ICat/src/CatalogDownloadDataFiles.cpp

```diff
@@ -164,10 +164,8 @@ std::string CatalogDownloadDataFiles::doDownloadandSavetoLocalDrive(
         nullptr, certificateHandler, context);
 
     // Session takes ownership of socket
-    Poco::Net::SecureStreamSocket *socket =
-        new Poco::Net::SecureStreamSocket(context);
-    Poco::Net::HTTPSClientSession session(*socket);
-    socket = nullptr;
+    Poco::Net::SecureStreamSocket socket(context);
```

Comment:
It doesn't look like the socket is being deleted, although I can't prove it without further investigation. I'll create a separate issue for this so it doesn't hold up this PR

Thanks for the tip on the tests - I hadn't noticed they were disabled. Unfortunately they crash when I enable them (not related to this change). I'll look into that as part of the separate issue.